File: UICPC/21/nwerc2020all/conteststruggles/submissions/accepted/paul.kt
private fun readInts() = readLine()!!.split(' ').map { it.toInt() }

fun main() {
    val (n, k) = readInts()
    val (a, s) = readInts()
    val res = (n*a-k*s).toDouble()/(n-k)
    println(if (res < 0 || res > 100) "impossible" else res)
}

File: UICPC/21/nwerc2020all/conteststruggles/submissions/accepted/paul.cpp
#include <cstdio>

int main() {
	int n, k;
	double a, s;
	scanf("%d%d%lf%lf", &n, &k, &a, &s);

	double res = (n*a-k*s)/(n-k);
	if (res < -1e-9 || res > 100+1e-9) printf("impossible\n");
	else printf("%.10f\n", (n*a-k*s)/(n-k));
}
